test/e2e/resource_deletion_protection_test.go

```diff
@@ -0,0 +1,87 @@
+package e2e
+
+import (
+	"context"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/test/e2e/framework"
+	"github.com/karmada-io/karmada/test/helper"
+)
+
+var _ = ginkgo.Describe("[resource deletion protection] deletion protection testing", func() {
+	var deploymentName, namespaceName string
+	var namespace *corev1.Namespace
+	var deployment *appsv1.Deployment
+
+	// update resource's label like this will protect the resource.
+	protectedLabelValues := map[string]string{
+		workv1alpha2.DeletionProtectionLabelKey: workv1alpha2.DeletionProtectionAlways,
+	}
+	noProtectedLabelValues := map[string]string{
+		workv1alpha2.DeletionProtectionLabelKey: "",
+	}
+
+	// create the deployment and namespaces for test.
+	ginkgo.BeforeEach(func() {
+		deploymentName = deploymentNamePrefix + rand.String(RandomStrLength)
+		namespaceName = "karmada-e2e-" + rand.String(RandomStrLength)
+		namespace = helper.NewNamespace(namespaceName)
+		deployment = helper.NewDeployment(namespaceName, deploymentName)
+		framework.CreateNamespace(kubeClient, namespace)
+		framework.CreateDeployment(kubeClient, deployment)
+	})
+
+	ginkgo.It("delete the protected deployment", func() {
```

Comment:
Test code blocks organized by `ginkgo.It` are previously independent of each other, they will be executed separately, that is to say, how many `ginkgo.It` blocks exist, how many test cases will be executed, from the description of your test, I think you are trying to organize these tests in a test case, you can place the steps that need to be tested in an` ginko.It` block, and then uses the `ginkgo.By` for code block organization.

Reply:
I want to have three separate test cases, so that in the presence of BeforeEach and AfterEach, I only need to focus on the content within a specific test case.

Reply:
Okay, I get it.